src/scheduler/simulate.cpp

```diff
@@ -1919,25 +1919,25 @@ policy_change_info(server_info *sinfo, resource_resv *resresv)
 		return 0;
 	}
 
-	if (sinfo->queues != NULL) {
+	if (!sinfo->queues.empty()) {
```

Comment:
I don't think you need this block any more.  It was there to stop dereferencing a NULL pointer.  The bottom 3 ifs will just fall through if sinfo->queues is empty.